### PR TITLE
Premature return causes the plugin to incorrectly identify fixtures as unused

### LINKED ISSUE #11: Tests without fixtures can cause false deadfixture positives
Nice plugin.

I stumbled across a glitch in --deadfixtures. If a test which does not use fixtures is found, then the fixtures used by subsequent tests are considered dead.

This test fails
```
def test_dont_list_fixture_used_after_test_which_does_not_use_fixtures(testdir, message_template):
    testdir.makepyfile("""
        import pytest
        @pytest.fixture()
        def same_file_fixture():
            return 1
            
        def test_no_fixture_used():
            assert True
        def test_simple(same_file_fixture):
            assert 1 == same_file_fixture
    """)

    result = testdir.runpytest('--dead-fixtures')
    message = message_template.format(
        'same_file_fixture',
        'est_dont_list_fixture_used_after_test_which_does_not_use_fixtures'
    )

    assert result.ret == 0
assert message not in result.stdout.str()
```
The fix seems simple and I will raise a pull request for your consideration

### PR DESCRIPTION
**Pull request by tawonga**
Fixes #11

### CODE REVIEW

**Comment by jllorencetti** [comment]
@tawonga thanks! :)
feel free to add yourself to the [authors](https://github.com/jllorencetti/pytest-deadfixtures/blob/master/AUTHORS.rst) file if you wish

**Comment by tawonga** [comment]
Done. Thank you.

File: tests/test_deadfixtures.py [hunk 1]
```diff
@@ -237,3 +237,29 @@ def test_conftest_fixture():
...
+            assert True
+
+        def test_simple(same_file_fixture):
+            assert 1 == same_file_fixture
+    """)
+
+    result = testdir.runpytest('--dead-fixtures')
+    message = message_template.format(
+        'same_file_fixture',
+        'test_dont_list_fixture_used_after_test_which_does_not_use_fixtures'
```
Comment (jllorencetti) [inline tests/test_deadfixtures.py]: the test name here should be the same as the "outer test", in this case `test_do_not_list_fixture_used_after_test_which_does_not_use_fixtures`

**Comment by tawonga** [comment]
Done. Good pickup - thanks.

The test function name is a bit of a mouthful. Hope it makes sense.

Would you like the bug-fix branch squashed?

**Comment by jllorencetti** [comment]
@tawonga no need for that. :)
Thank you very much.

**Pull request merged**